hs_core/hydroshare/hs_bagit.py

```diff
@@ -41,217 +50,104 @@ def delete_files_and_bag(resource):
-            a.add_resource(ar)
-
-    # Register a serializer with the aggregation, which creates a new ResourceMap that needs a URI
-    serializer = RdfLibSerializer('xml')
-    resMap = a.register_serialization(serializer, res_map_url)
-    resMap._dc.identifier = resource.short_id
-
-    # Fetch the serialization
-    remdoc = a.get_serialization()
-
-    # change the namespace for the 'creator' element from 'dcterms' to 'dc'
-    xml_string = remdoc.data.replace('dcterms:creator', 'dc:creator')
-
-    # delete this extra element
-    # <ore:aggregates rdf:resource="[hydroshare domain]/terms/[Resource class name]"/>
-    xml_string = xml_string.replace(
-        '<ore:aggregates rdf:resource="%s"/>\n' % str(resource.metadata.type.url), '')
-
-    # create resourcemap.xml and upload it to iRODS
-    from_file_name = os.path.join(temp_path, 'resourcemap.xml')
-    with open(from_file_name, 'w') as out:
-        out.write(xml_string)
-    to_file_name = os.path.join(resource.root_path, 'data', 'resourcemap.xml')
-    istorage.saveFile(from_file_name, to_file_name, False)
-
-    res_coll = resource.root_path
-    istorage.setAVU(res_coll, 'metadata_dirty', "false")
-    shutil.rmtree(temp_path)
-    return istorage
+    bagdir = os.path.join(tmpdir, "bag")
+    os.makedirs(bagdir)
 
+    # make the bdbag and create a zip archive
+    bdb.make_bag(bagdir, checksums, False, False, False, None, metadata_file, remote_manifest_file, 'config/bdbag.json')
+    zipfile = bdb.archive_bag(bagdir, "zip")
 
-def create_bag(resource):
-    """
-    Modified to implement the new bagit workflow. The previous workflow was to create a bag from
-    the current filesystem of the resource, then zip it up and add it to the resource. The new
-    workflow is to delegate bagit and zip-up operations to iRODS, specifically, by executing an
-    iRODS bagit rule to create bagit file hierarchy, followed by execution of an ibun command to
-    zip up the bagit file hierarchy which is done asychronously as a celery task. This function
-    only creates all bag files under resource collection in iRODS and set bag_modified iRODS AVU
-    metadata to true so that the bag will be created or recreated on demand when it is being
-    downloaded asychronously by a celery task.
+    # save the zipped bag to iRODS for retrieval upon download request
+    istorage = resource.get_irods_storage()
+    bag_full_name = 'bags/{res_id}.zip'.format(res_id=resource.short_id)
+    irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    destbagfile = os.path.join(irods_dest_prefix, bag_full_name)
 
-    Parameters:
-    :param resource: (subclass of AbstractResource) A resource to create a bag for.
-    :return: the hs_core.models.Bags instance associated with the new bag.
-    """
-    create_bag_files(resource)
+    # delete any existing bag files
+    if (istorage.exists(destbagfile)):
+        istorage.delete(destbagfile)
```

Comment:
This is not needed. istorage.saveFile() will use -f flag to override existing bag file as needed.